### ISSUE TITLE
[Bug]: Date Picker and DetailList are not Accessible. (Scanned with Accessibility insights for Web)

### ISSUE THREAD

**Issue by ckibuchi** [microsoft/fluentui#28066]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

None

### Reproduction

React / v8 - https://aka.ms/fluentpen

## Actual Behavior
Date Picker and DetailList are failing Fast Pass

## Expected Behavior
It is expected that these components should pass FastPass

### Requested priority

High

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by Hotell** [comment]
can you please provide what particular a11y issues are present within fast pass ? without that information this will be considered as no repro - issue will be closed. thank you

**Comment by jongonzamsft** [comment]
<img width="405" alt="image" src="https://github.com/microsoft/fluentui/assets/41756907/947b745f-3419-41ae-9c16-effaf485ea1a">
<img width="510" alt="image" src="https://github.com/microsoft/fluentui/assets/41756907/0c878482-7f2d-44e2-bfed-77a9fce044a2">

Title: WCAG 1.3.1: Ensures elements with an ARIA role that require child roles contain them (table)
Tags: Accessibility, WCAG 1.3.1, aria-required-children

Issue: Ensures elements with an ARIA role that require child roles contain them (aria-required-children - https://accessibilityinsights.io/info-examples/web/aria-required-children)

Target application: Fluent UI - Controls - React - DatePicker - https://developer.microsoft.com/en-us/fluentui#/controls/web/datepicker

Element path: table

Snippet: <table class="table-448" aria-multiselectable="false" aria-labelledby="id__1854" aria-activedescendant="id__1855" role="grid">

How to fix: 
Fix any of the following:
  Element has children which are not allowed (see related nodes)
  Element has no aria-busy="true" attribute

Environment: Microsoft Edge version 114.0.1823.43

====

This accessibility issue was found using Accessibility Insights for Web 2.38.1 (axe-core 4.6.3), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

**Comment by ckibuchi** [comment]
Exactly the above error and description given is the same I used.
Both for Details List and Date Picker
